[team-05] FE 초기 환경 설정

bread1022 opened the pull request:
안녕하세요 제이미 😊 !
Second-hand 리뷰를 받게 된 @ahnlook, @bread1022 입니다 !!
두번째 프로젝트인만큼 설계나 초기환경 설정은 빠르게 진행될 줄 알았는데 역시나 쉽지 않은 한 주였네요...
우선 이번 첫주차에 저희는 협업 지향점에 대해 서로 이야기해보면서 FE 내부 코딩 컨벤션과 페어프로그래밍을 위한 그라운드룰 등을 정하였고
프로젝트 환경으로 React + vite + Typescript 를 설정하였습니다! 앞으로 잘 부탁드립니다 :)

## 💻 구현내용 및 진행상황
- 페어프로그래밍을 위한 [그라운드룰](https://github.com/masters2023-2nd-project-05/second-hand/wiki/FE-%EA%B7%B8%EB%9D%BC%EC%9A%B4%EB%93%9C%EB%A3%B0)을 정하였습니다.
- [코딩컨벤션](https://github.com/masters2023-2nd-project-05/second-hand/wiki/FE-Coding-Convention)을 작성하였습니다.
- [디렉토리구조](https://github.com/masters2023-2nd-project-05/second-hand/wiki/Directory-structure)를 설계하였습니다.
- 백로그를 작성하고 BE,iOS팀과의 협의를 통해 [API 명세](https://second-hand-team5.notion.site/API-05a5c9b2f7b94920bcc060d6cd4ac080)를 완료하였습니다.
- closed: https://github.com/masters2023-2nd-project-05/second-hand/issues/6
- closed: https://github.com/masters2023-2nd-project-05/second-hand/issues/30
- closed: https://github.com/masters2023-2nd-project-05/second-hand/issues/24

### Vite를 사용하게 된 이유
이전의 CRA 번들러는 파일 하나가 수정되면 전체를 번들링하지만 Vite의 경우 entry에서 수정된 모듈과 관련된 부분만 번들링을 진행하다보니 속도면에 빠른 장점을 가지고 또 마스터분들이 언급을 몇번 해주셔서 관심을 갖게되어 이번 프로젝트에 한번 도전해보고 싶었습니다.

## 🤔 고민이나 해결과정

### 릴리

- 나니와 협업 지향점을 1명이 작성한 것처럼 코드를 작성하자인데, 현업에는 코딩 컨벤션 등 다양한 협업 설정이 되어 있어서 1명이 작성한 것 같은 코드처럼 보이나요?
- webpack 대신 vite를 도입해봤는데 환경 설정부터 관문이 많았던 것처럼 느껴졌어요. 저희가 취업을 하게 되면 실제로 webpack, vite를 직접 설정할 일은 없겠지만 그래도 하나를 익혀야 한다면 제이미는 어떤 것을 추천하시나요?
- UI test tool로 storybook을 도입했는데 현업에서 많이 사용하는 추세인가요? 너무 귀찮은 테스트 작업인데 막상 하고나면 편하고 좋아서 항상 빼야할지에 대한 고민이 많은 것 같아요.

### 나니

- React + Typescript 도 처음이지만 Vite 까지 적용하려니 초기 환경 설정에 어려움을 느낄 때마다 ... 과연 어디서 부터 잘 못 된 걸까 싶을 정도로 저번 미션때 마주하지 못한 에러들이 초기 환경 설정 때부터 발생했었는데요... 현업에서 vite 에 대한 인식은 어떤지 궁금합니다 !

## ☑️ 리뷰포인트
- `GlobalStyle`로 page size를 `.app`에 주었는데 body나 App.tsx에 직접 스타일 값을 주는 것은 좋지 않다고 들은 것 같아요. page layout을 만들어서 스타일 값을 주는 것이 좋을까요?
- `GlobalProvider` 파일을 만들까에 대한 고민이 드는데 provider의 depth가 어느 정도 깊어지면 분리한다에 대한 기준이 제이미는 있으신가요?

bread1022 committed:
[#6] CHORE: project 환경설정

- React + vite + TS 를 설치하였습니다.
- 스타일 컴포넌트를 설치하였습니다.
- eslint를 설치하였습니다.

bread1022 committed:
[#6] CHORE: project 환경설정(prettier)

- prettier를 설정하였습니다.

bread1022 committed:
[#6] CHORE: project 환경설정(eslint)

- 중복되는 eslint.cjs 파일을 제거하였습니다.

bread1022 committed:
[#6] MERGE: project-init

MERGE: Fe/006 project init

ahnlook committed:
[#6] CHORE: eslint import 플러그인 추가

ahnlook committed:
Merge pull request #17 from masters2023-2nd-project-05/FE/006-set-init

[#6] CHORE: eslint import 플러그인 추가

ahnlook committed:
[#6] CHORE: css global style 적용

ahnlook committed:
[#6] CHORE: 불필요한 파일 삭제

ahnlook committed:
[#6] CHORE: css 초기 변수 설정

ahnlook committed:
[#6] CHORE: 디렉토리 구조 설계

- 공통 컴포넌트 폴더 생성하였습니다.

ahnlook committed:
[#6] CHORE: 공통 컴포넌트 index 파일 생성

bread1022 committed:
[#24] FEAT: 로그인 페이지 로그인 기능 구현

- 로그인 페이지에서 임시로 로그인한 뒤 유저의 정보를 저장하였습니다.
- favicon을 추가하였습니다.
- svgs 폴더를 생성하였습니다.

Refs: #6

bread1022 committed:
MERGE: 024-Login-UI

[#24] FEAT: 로그인 페이지 로그인 기능 구현 (페어로 진행)

ahnlook committed:
CHORE: yarn.lock 파일 삭제

bread1022 committed:
[#6] CHORE: Typescript 적용을 위한 스타일 변수 설정

bread1022 committed:
[#6] CHORE: 스타일 변수 수정

- default Props error 를 해결하였습니다.
- svg 파일의 fill 속성을 current로 변경하였습니다.

bread1022 committed:
MERGE: 006-theme

[#6] CHORE: 스타일 변수 수정

bread1022 committed:
[#6] CHORE: font 변수 내용 수정

bread1022 committed:
[#35] FEAT: Icon 컴포넌트 생성

- svg 파일의 아이콘을 사용하기 위해 플러그인 설치 후 컴포넌트를 생성하였습니다.

ahnlook committed:
Merge: 05/FE/035-ICON

[#35] FEAT: Icon 컴포넌트 생성

ahnlook committed:
[#6] CHORE: 절대 경로 설정

bread1022 committed:
[#35] ADD: svg 파일 크기 수정

- width, heigth value를 current로 수정하였습니다.

bread1022 committed:
[#35] ADD: icon size 수정

- width, height 를 지정하지 않고 icon size를 입력받아 아이콘 크기를 설정하게끔 구현하였습니다.

mina-gwak commented on front-end/package.json:
```diff
@@ -0,0 +1,54 @@
+    "build-storybook": "storybook build"
+  },
+  "dependencies": {
+    "@types/styled-components": "^5.1.26",
```
설치한 패키지를 보면 dependency와 devDependency의 차이에 대해서는 알고 계신듯 한데 아마 이 친구만 `dependencies`에 포함되어 있는 건 설치 중 실수로 그러신 거겠죠? `devDependencies`로 옮기는 게 좋겠네요!

mina-gwak commented on front-end/package.json:
```diff
@@ -0,0 +1,54 @@
+    "@vitejs/plugin-react": "^4.0.0",
+    "eslint": "^8.0.1",
+    "eslint-config-prettier": "^8.8.0",
+    "eslint-config-standard-with-typescript": "^35.0.0",
```
오호 처음 보는 패키지라서 찾아보니 `eslint-config-airbnb`, `eslint-config-google` 이런 것들과 같은 결인가 보네요. 이걸 선택하신 이유가 궁금해요!

ahnlook replied:
eslint init시 선택했던 것들 (react, ts, standard)에 의해 자동 설치되었던 것 같아요!
extends에 추가를 안 해놓은 거보니 사용을 안 하는 것 같아서 나니와 상의 후에 적용 또는 삭제하겠습니다!

mina-gwak replied:
좋습니다 👍 여러 스타일 가이드가 있으니 취향에 맞는 게 있다면 그 중에서 골라보셔도 괜찮을 것 같아요!

mina-gwak commented on front-end/package.json:
```diff
@@ -0,0 +1,54 @@
+    "prettier": "^2.8.8",
+    "prop-types": "^15.8.1",
+    "storybook": "^7.0.18",
+    "typescript": "*",
```
앗 패키지 버전을 `*`로 지정하시는 건 좀 위험해보이는데요..?
`package.json`이 어떤 역할을 하는지, semantic versioning에 대해서 찾아보시고 수정해보시면 좋을 것 같아요.

mina-gwak commented on front-end/src/App.tsx:
```diff
@@ -0,0 +1,28 @@
+import Login, { UserInfo } from './pages/Login';
+import GlobalStyle from './styles/globalStyles';
+import theme from './styles/theme';
+export const UserInfoContext = createContext({});
```
아직 구조를 잡아가는 중이라 그러실 수도 있겠지만 컨텍스트는 따로 분리해서 사용해도 좋을 것 같아요. 참고하실만한 글 두고 갑니다!

https://kentcdodds.com/blog/how-to-use-react-context-effectively

mina-gwak commented on front-end/src/App.tsx:
```diff
@@ -0,0 +1,28 @@
+export const UserInfoContext = createContext({});
+
+function App() {
+  const [userInfo, setUserInfo] = useState({});
```
`useState`를 사용할 때 값이 단순히 `string`, `number` 정도라면 초기값을 보고 유추가 가능하겠지만 값이 특정 몇가지로 한정되는 경우라던가 지금처럼 객체인 경우에는 타입을 지정해주면 좋습니다! 어떤 값인지 명확히 알 수 있으니까요.

ahnlook replied:
```tsx
const [userInfo, setUserInfo] = useState<UserInfo>({
  id: 0,
  memberID: '',
  profileImgUrl: '',
  region: []
});
```

제이미 말씀대로 위와 같의 정의해 두려고 합니다!
각 value에 null을 넣는 것보다 타입에 맞게 초기화하는 것이 좋을까요?

mina-gwak replied:
제 생각에는 각 상황에 따라 다를 것 같아요! 초기값이 필요하다거나 하는 경우에는 작성하신 것처럼 할 수도 있겠고, `userInfo && <UserProfile />`처럼 조건부 렌더링을 한다거나 로딩 처리를 한 다음 데이터를 받아온 후에 페이지를 보여준다고 하면 굳이 필요하지 않을 수도 있겠네요.

mina-gwak commented on front-end/src/App.tsx:
```diff
@@ -0,0 +1,28 @@
+  };
+
+  return (
+    <>
+      <ThemeProvider theme={theme}>
+        <UserInfoContext.Provider value={{ userInfo, handleUserInfo }}>
+          <GlobalStyle />
+          <div className="app">
+            <Login handleUserInfo={handleUserInfo} />
+          </div>
+        </UserInfoContext.Provider>
+      </ThemeProvider>
+    </>
```
1. 여기서 질문 하나 드릴게요! 아래 두 가지 코드는 차이점이 있을까요? 있다면 어떻게 다를까요?

```tsx
const Context = useContext();

const App = () => {
  const [state, setState] = useState(false);

  return (
    <Context.Provider value={state}>
      <Component />
    </Context.Provider>
  );
}
```

```tsx
const Context = useContext();

const ContextProvider = ({ children }) => {
  const [state, setState] = useState(false);

  return (
    <Context.Provider value={state}>
      {children}
    </Context.Provider>
  );
}

const App = () => {
  return (
    <ContextProvider>
      <Component />
    </ContextProvider>
  );
}
```

2. 컨텍스트를 사용하는 경우 `Login` 컴포넌트에서는 컨텍스트에서 값을 가져와서 사용할 수 있겠죠? props로 넘겨줄 필요는 없어보이네요!

bread1022 replied:
넵 ! 컨벤션 맞추는 과정에서 테스트로 작성해본 코드였는데 참고해서 수정해볼게요 ! 🥲

mina-gwak commented on front-end/src/assets/Icon.tsx:
```diff
@@ -0,0 +1,29 @@
+import * as iconTypes from '@assets/svgs/index';
+import palette from '@styles/colors';
```
`@assets` 와 같은 방식이면 절대경로를 설정해서 사용하시는 건가 싶은데 설정하신 부분이 보이지는 않네요? 🤔

bread1022 replied:
절대 경로도 수정 중이었어서 커밋을 안했었는데, 현재 설정 완료되어 커밋 하였습니다 !

mina-gwak commented on front-end/src/assets/Icon.tsx:
```diff
@@ -0,0 +1,29 @@
+  xl: 30,
+};
+
+const Icon = ({
+  name,
+  size = 'md',
+  fill = palette.neutral.text,
+}: IconProps) => {
+  const IconComponent = iconTypes[name];
+  const iconSize = iconSizes[size];
+
+  return <IconComponent width={iconSize} height={iconSize} fill={fill} />;
+};
```
수정하시기 전에 잠깐 봤을 때는 `width`, `height`로 받았던 걸로 기억하는데 사이즈를 받는 걸로 변경하셨더라구요. 지금 방식이 훨씬 좋아보여요! 👍

그리고 이 친구도 컴포넌트라고 볼 수 있을 것 같은데, `components`로 옮겨도 되지 않나 하는 생각이 드네요.

mina-gwak commented on front-end/src/pages/Login.tsx:
```diff
@@ -0,0 +1,57 @@
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+export interface UserInfo {
+  id: number;
+  memberId: string;
+  profileImgUrl: string;
+  region: number[];
+}
```
요 부분이 `userInfo` 상태의 타입으로 가면 되겠네요!

mina-gwak commented on front-end/src/pages/Login.tsx:
```diff
@@ -0,0 +1,57 @@
+const Login = ({ handleUserInfo }: LoginProps) => {
+  const [inputValue, setInputValue] = useState('');
+
+  const fetchPost = async (inputValue: string) => {
+    const navigate = useNavigate();
+    const res = await fetch('/login', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ memberId: inputValue }),
+    });
+    const data = await res.json();
+    if (data) {
+      handleUserInfo(data);
+      navigate('/');
+    }
+  };
```
이 함수가 하는 일을 보면 크게 데이터를 가져오는 것, 데이터를 가져온 이후의 처리 (상태 변경, 페이지 이동) 이렇게 두 가지로 나눌 수 있을 것 같은데 후자의 경우 `fetchPost`라는 이름만 봤을 때 이 안에서 처리될 것이라고 예측하기 어려워 보여요. 그렇다면 이 함수가 의도한 것보다 많은 일을 처리하고 있다는 뜻이겠죠? 함수가 하는 일이 명확해지도록 분리하면 좋을 것 같네요!

추가로 fetch하는 로직은 반복되어 사용될 수 있으니 따로 분리해서 사용하시는 것도 고려해보세요! (티타임 때 말씀하셨던 커스텀 훅으로 분리할 수 있는 경우가 되겠네요 😁)

ahnlook replied:
함수, 커스텀 훅으로 분리해보겠습니다!

mina-gwak commented on front-end/src/pages/Login.tsx:
```diff
@@ -0,0 +1,57 @@
+
+export default Login;
+
+const MyLogin = styled.div``;
```
아직 제대로 된 스타일 작업을 시작하시기 전이라서 요런 상태로 있는 거겠죠? 🤔

ahnlook replied:
제이미 리뷰를 보니 주석으로 TODO 남겨 두었어야겠다 싶네요! 앞으로 신경써서 올리겠습니다!

mina-gwak commented on front-end/src/styles/fonts.ts:
```diff
@@ -0,0 +1,48 @@
+    fontWeight: 400,
+    fontSize: '28px',
+  },
+  title2: {
+    fontWeight: 400,
+    fontSize: '22px',
+  },
+  title3: {
+    fontWeight: 400,
+    fontSize: '20px',
+  },
+  headline: {
+    fontWeight: 600,
+    fontSize: '17px',
+  },
+  body: {
+    fontWeight: 400,
+    fontSize: '17px',
+  },
+  callout: {
+    fontWeight: 400,
+    fontSize: '16px',
+  },
+  subhead: {
+    fontWeight: 400,
+    fontSize: '15px',
+  },
+  footnote: {
+    fontWeight: 400,
+    fontSize: '13px',
+  },
+  caption1: {
+    fontWeight: 400,
+    fontSize: '12px',
+  },
+  caption2: {
+    fontWeight: 400,
+    fontSize: '11px',
+  },
+};
```
흐음 의문이 드는 부분이 두 가지 있는데요!

1. 사용되는 케이스가 새롭게 생길 때마다 매번 프로퍼티를 추가해주면 너무 많아지지는 않을까요?
2. 만약 `title`이라고 네이밍된 스타일을 제목과 같은 요소가 아닌 곳에서 사용하면 나중에 봤을 때 의문이 들지 않을까요?

ahnlook replied:
전역 변수로 선언해두는 방법 외에 좋은 방법을 찾지 못했던 것 같아요 :(
혹시 현업에서는 어떻게 관리하는지 좋은 방법이 있을까요?

(질문) figma에 typography로 정의된 부분을 코드로 정의해 두었는데, 이런 디자인 시스템(?) 적인 부분이 새롭게 생기는 경우가 많이 있는지도 궁금해요!

mina-gwak replied:
아하 피그마에 작성된 스타일을 옮기셨던 거군요! 그렇다면 새로 추가될 것에 대한 염려는 하지 않아도 되겠네요. 저도 스타일을 이렇게 모아서 관리하는 것 자체는 괜찮다고 생각하는데, 위에서 말씀드렸던 2번처럼 쓰이는 상황, 요소에 따른 네이밍을 하는 데 있어서 의도된 상황에서 사용되지 않는 경우가 생긴다면 그게 괜찮을까? 싶은 생각에서 말씀드렸었어요. 

새롭게 생기는 경우가 많냐는 질문에 대해서는 확실하게 체계가 갖추어져있다면 변경이 적을 거고, 체계를 갖춰나가는 중이라면 새롭게 추가된다거나 변경되는 경우가 많을테니 이것도 상황에 따라 다르지 않을까 하는 생각입니다!

mina-gwak commented on front-end/package.json:
```diff
@@ -0,0 +1,54 @@
+    "eslint-plugin-react-refresh": "^0.3.4",
+    "eslint-plugin-storybook": "^0.6.12",
+    "prettier": "^2.8.8",
+    "prop-types": "^15.8.1",
```
타입스크립트를 사용하고 계시니 `prop-types`는 사용할 일이 없지 않을까요? 사용하지 않는 패키지는 제거해주세요!

mina-gwak reviewed (approved):
릴리, 나니 안녕하세요! 두분의 리뷰를 맡은 제이미입니다 😀

환경설정은 언제나 쉽지 않죠 🥲 거기다 vite를 사용하셨다니 더 그러셨겠어요.. 저도 이전에 잠깐 vite로 프로젝트 세팅을 해볼 일이 있었는데 많이 헤맸었던 기억이 있어요. 그래도 이왕 도전해보시기로 한 거 잘 헤쳐나가시길 바라겠습니다!

<br/>

그럼 작성해주신 질문에 대해 답변 드려볼게요.

<br/>

> 나니와 협업 지향점을 1명이 작성한 것처럼 코드를 작성하자인데, 현업에는 코딩 컨벤션 등 다양한 협업 설정이 되어 있어서 1명이 작성한 것 같은 코드처럼 보이나요?

컨벤션이 잘 작성되어있더라도 코드의 세부적인 부분까지 모두 규정할 수는 없기 때문에 말씀하신 것처럼 1명이 작성한 것처럼 보이기는 어렵지 않을까 싶어요. 하지만 목표 자체는 좋다고 생각합니다! 따로 PR 리뷰도 하고 계신 것 같던데 페어 프로그래밍과 리뷰 등을 통해 많이 이야기를 나눠보시면서 두분만의 컨벤션을 만들어나가면 그래도 목표에 가까워질 수 있지 않을까요? 컨벤션을 정한 후에도 의식적으로 컨벤션을 지키도록 하는 노력도 필요하고, ESLint나 Prettier 설정을 통해 컨벤션을 강제하는 방안도 있으니 점차적으로 적용해나가보시면 될 것 같아요. 
추가로 설정한 ESLint, Prettier 규칙을 특정 시점에 자동으로 적용하도록 할 수 있는 `husky`라는 것도 있는데, 시간 되실 때 한번 찾아보셔도 좋을 것 같네요!

<br/>

> webpack 대신 vite를 도입해봤는데 환경 설정부터 관문이 많았던 것처럼 느껴졌어요. 저희가 취업을 하게 되면 실제로 webpack, vite를 직접 설정할 일은 없겠지만 그래도 하나를 익혀야 한다면 제이미는 어떤 것을 추천하시나요?

> React + Typescript 도 처음이지만 Vite 까지 적용하려니 초기 환경 설정에 어려움을 느낄 때마다 ... 과연 어디서 부터 잘 못 된 걸까 싶을 정도로 저번 미션때 마주하지 못한 에러들이 초기 환경 설정 때부터 발생했었는데요... 현업에서 vite 에 대한 인식은 어떤지 궁금합니다 !

일단 제 생각을 먼저 말씀드리자면 (정답은 아닙니다!) vite 자체는 좋은 도구라고 생각해요! 잠깐 사용해봤을 때 확실히 빌드 속도가 빨랐던 것으로 기억하구요. 제대로 프로젝트를 해보지는 않아서 확실히는 모르겠지만, 프로젝트를 진행하는 데에 문제가 없다면 CRA 대신 vite를 적용해봐도 좋을 것 같다는 생각입니다.
학습적인 측면에서는 꼭 둘 중에 하나만 먼저 학습해야 한다면 webpack 쪽이지 않을까 싶어요. 아직까지는 vite를 사용하고 있는 회사가 그렇게 많지 않을 거라고 생각돼서 취업을 하신 후에 설정을 만질 일이 있다면 vite보다는 webpack일 확률이 좀 더 높을 거라고 생각합니다. 하지만 이미 설정을 해두신 지금 상황에서는 이후에 공부할 기회는 얼마든지 있으니 이번 프로젝트에서는 계속 vite로 진행해보셔도 좋을 것 같아요! 말씀하신 것처럼 다양한 에러를 마주하는 경험도 나중에 분명 도움이 되실 거예요 🙂

<br/>

> UI test tool로 storybook을 도입했는데 현업에서 많이 사용하는 추세인가요? 너무 귀찮은 테스트 작업인데 막상 하고나면 편하고 좋아서 항상 빼야할지에 대한 고민이 많은 것 같아요.

저희는 현재 사용하고 있지는 않지만 한번 사용해보자고 말씀드려서 하반기에 도입을 해 볼 예정입니다! 저 같은 경우에는 말씀하신대로 좀 귀찮을 수도 있는 작업이지만 컴포넌트 개발할 때나 PR 작성 시 수정된 부분을 확인한다거나 하는 경우에 편리했다고 느꼈던 경험이 있었거든요. 이번 기회에 한번 잘 활용해보시면 어떨까요?

<br/>

> GlobalStyle로 page size를 .app에 주었는데 body나 App.tsx에 직접 스타일 값을 주는 것은 좋지 않다고 들은 것 같아요. page layout을 만들어서 스타일 값을 주는 것이 좋을까요?

넵, 저도 후자가 더 나을 것 같다고 생각해요!

<br/> 

> GlobalProvider 파일을 만들까에 대한 고민이 드는데 provider의 depth가 어느 정도 깊어지면 분리한다에 대한 기준이 제이미는 있으신가요?

GlobalProvider 라고 말씀해주셔서 조금 헷갈리는데 context를 생성해서 사용하는 기준에 대해 말씀하신 걸까요 아니면 다른 부분에 대한 이야기일까요? 좀 더 설명을 덧붙여주시면 감사하겠습니다!

<br/>

질문에 답변드리다보니 코멘트가 좀 길어졌네요 😅 리뷰도 그렇고 제가 말씀드리는 게 정답은 아니니 모두 받아들이실 필요는 없습니다! 질문이 있으시면 언제든지 코멘트나 디엠 남겨주세요. 한 주 동안 고생 많으셨습니다!

ahnlook committed:
[#6] eslint, prettier 설정

- eslint에 import 룰 등을 추가하였습니다.
- prettier 상세 설정을 하였습니다.

bread1022 committed:
MERGE: 006-set-ts

[#6] CHORE: 절대 경로 설정

ahnlook commented:
@mina-gwak 제이미! `globalProvider` 부분에 대해 다시 말씀드립니당!
```tsx
// App.tsx
// ...
    <>
      <ThemeProvider theme={theme}>
        <UserInfoContext.Provider value={{ userInfo, handleUserInfo }}>
          <GlobalStyle />
          <div className="app">
            <Login handleUserInfo={handleUserInfo} />
          </div>
        </UserInfoContext.Provider>
      </ThemeProvider>
    </>
```

위 부분에서 provider을 파일(`globalProvider.tsx`)을 분리해서
아래와 같이 `index.tsx`에 provider들을 적용하려고 합니다!

```tsx
const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement)
root.render(
  <React.StrictMode>
    <GlobalProvider>
      <RouterProvider router={router} />
    </GlobalProvider>
  </React.StrictMode>
)
```

이 때 provider의 depth가 어느 정도일 때 분리를 하는 것이 좋을지에 대한 고민이었습니다 :)

bread1022 committed:
[#6] ADD: svg 파일 사용을 위한 환경 설정

- vite plugin react svg 셋팅하였습니다.
- typscript 버전을 설정하였습니다.
- 절대경로에 따른 import eslint 설정을 수정하였습니다.
